Sources/ContextPropagation/InstrumentationMiddleware.swift

```diff
@@ -0,0 +1,28 @@
+public protocol InstrumentationMiddlewareProtocol {
+    associatedtype ExtractFrom
+    associatedtype InjectInto
+
+    func extract(from: ExtractFrom, into context: inout Context)
+    func inject(from context: Context, into: inout InjectInto)
+}
+
+public struct InstrumentationMiddleware<InjectInto, ExtractFrom>: InstrumentationMiddlewareProtocol {
+    private let extract: (ExtractFrom, inout Context) -> Void
+    private let inject: (Context, inout InjectInto) -> Void
+
+    public init(
+        extract: @escaping (ExtractFrom, inout Context) -> Void,
+        inject: @escaping (Context, inout InjectInto) -> Void
+    ) {
+        self.extract = extract
+        self.inject = inject
+    }
+
+    public func extract(from: ExtractFrom, into context: inout Context) {
+        self.extract(from, &context)
+    }
+
+    public func inject(from context: Context, into: inout InjectInto) {
+        self.inject(context, &into)
+    }
+}
```

Comment:
kk, good move :) We'll see if needs completely separate or here is fine in the long run, but now definitely good here since all use case will share it 👍

Reply:
Yes, I also briefly considered created a separate library for that but figured that we'll get around to that eventually once we gathered more insights ✅